src/QsCompiler/Tests.Compiler/CompletionParsingTests.fs

```diff
@@ -2,185 +2,205 @@
+    let test = testAs NamespaceTopLevel
+    let characteristics = [Keyword "Adj"; Keyword "Ctl"]
     test "operation " [Declaration]
     test "operation Foo" [Declaration]
     test "operation Foo " []
     test "operation Foo (" [Declaration]
     test "operation Foo (q" [Declaration]
-    test "operation Foo (q :" [Type]
-    test "operation Foo (q : Qubit" [Type]
+    test "operation Foo (q :" types
+    test "operation Foo (q : Qubit" types
     test "operation Foo (q : Qubit)" []
-    test "operation Foo (q : Qubit) :" [Type]
-    test "operation Foo (q : Qubit) : " [Type]
-    test "operation Foo (q : Qubit) : Unit" [Type]
+    test "operation Foo (q : Qubit) :" types
+    test "operation Foo (q : Qubit) : " types
+    test "operation Foo (q : Qubit) : Unit" types
     test "operation Foo (q : Qubit) : Unit " [Keyword "is"]
     test "operation Foo (q : Qubit) : Unit i" [Keyword "is"]
     test "operation Foo (q : Qubit) : Unit is" [Keyword "is"]
-    test "operation Foo (q : Qubit) : Unit is " [Characteristic]
-    test "operation Foo (q : Qubit) : Unit is A" [Characteristic]
-    test "operation Foo (q : Qubit) : Unit is Adj" [Characteristic]
+    test "operation Foo (q : Qubit) : Unit is " characteristics
+    test "operation Foo (q : Qubit) : Unit is A" characteristics
+    test "operation Foo (q : Qubit) : Unit is Adj" characteristics
     test "operation Foo (q : Qubit) : Unit is Adj " []
-    test "operation Foo (q : Qubit) : Unit is Adj +" [Characteristic]
-    test "operation Foo (q : Qubit) : Unit is Adj + " [Characteristic]
-    test "operation Foo (q : Qubit) : Unit is Adj + Ctl" [Characteristic]
-    test "operation Foo (q : Qubit) : Unit is (" [Characteristic]
-    test "operation Foo (q : Qubit) : Unit is ( " [Characteristic]
-    test "operation Foo (q : Qubit) : Unit is (Adj" [Characteristic]
-    // TODO: test "operation Foo (q : Qubit) : Unit is (Adj " []
-    test "operation Foo (q : Qubit) : Unit is (Adj + " [Characteristic]
-    test "operation Foo (q : Qubit) : Unit is (Adj + Ctl" [Characteristic]
-    // TODO: test "operation Foo (q : Qubit) : Unit is (Adj + Ctl)" []    
+    test "operation Foo (q : Qubit) : Unit is Adj +" characteristics
+    test "operation Foo (q : Qubit) : Unit is Adj + " characteristics
+    test "operation Foo (q : Qubit) : Unit is Adj + Ctl" characteristics
+    test "operation Foo (q : Qubit) : Unit is (" characteristics
+    test "operation Foo (q : Qubit) : Unit is ( " characteristics
+    test "operation Foo (q : Qubit) : Unit is (Adj" characteristics
+    test "operation Foo (q : Qubit) : Unit is (Adj " []
+    test "operation Foo (q : Qubit) : Unit is (Adj + " characteristics
+    test "operation Foo (q : Qubit) : Unit is (Adj + Ctl" characteristics
+    test "operation Foo (q : Qubit) : Unit is (Adj + Ctl)" []    
     test "operation Foo (q : Qubit) : Unit is Adj + Ctl " []
-    test "operation Foo (q : Qubit) : Unit is Adj + Cat " []
```

Comment:
I changed the parser to throw an exception when parsing fails, instead of returning an empty list. Instead of changing this test to use `Assert.Throws` I just removed it. :) But now I changed the testing setup to make it easier to have tests for both successes and failures, and I added this test and the other two back.

Reply:
Ok. Why did you choose to go with throwing opposed to simply making the parsing fail?

Reply:
Ok, got it reading the comment below. See my reply below.